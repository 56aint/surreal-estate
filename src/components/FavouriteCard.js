import React from "react";
import PropTypes from "prop-types";
import { FaTrash } from "react-icons/fa";
import "../styles/PropertyCard.css";

const FavouriteCard = ({ _id, title, city, onDeleteFavourite }) => {
  return (
    <div className="favourite-card">
      <div className="favourite-card_title" data-testid="fav_title-id">
        {title}
      </div>
      <div className="favourite-card_city" data-testid="fav_city-id">
        {city}
      </div>
      <button
        href="#"
        className="delete-button"
        type="button"
        onClick={() => onDeleteFavourite(_id)}
      >
        <FaTrash className="delete-sign" />
        Remove
      </button>
    </div>
  );
};

FavouriteCard.propTypes = {
  title: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  onDeleteFavourite: PropTypes.func.isRequired,
  _id: PropTypes.string.isRequired,
};

export default FavouriteCard;
