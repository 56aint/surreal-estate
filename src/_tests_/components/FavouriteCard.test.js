import React from "react";
import { render } from "@testing-library/react";
import FavouriteCard from "../../components/FavouriteCard";

describe("FavouriteCard", () => {
  const mockFavourites = {
    _id: "id",
    title: "1 bedroom house",
    type: "Detached",
    city: "Brighton",
    onDeleteFavourite: jest.fn(),
  };
  it("renders correctly", () => {
    const { asFragment } = render(<FavouriteCard {...mockFavourites} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct props", () => {
    const { getByText, getByTestId, getByRole } = render(
      <FavouriteCard {...mockFavourites} />
    );
    expect(getByTestId("fav_title-id")).toHaveClass("favourite-card_title");
    expect(getByTestId("fav_city-id")).toHaveClass("favourite-card_city");
    expect(getByText("Remove")).toHaveClass("delete-button");
    expect(getByText("Remove")).toBeTruthy();
    expect(getByRole("button")).toBeTruthy();
    expect(getByRole("button")).toHaveClass("delete-button");
  });
});
