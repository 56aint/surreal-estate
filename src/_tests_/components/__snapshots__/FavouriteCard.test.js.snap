// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="favourite-card"
  >
    <div
      class="favourite-card_title"
      data-testid="fav_title-id"
    >
      1 bedroom house
    </div>
    <div
      class="favourite-card_city"
      data-testid="fav_city-id"
    >
      Brighton
    </div>
    <button
      class="delete-button"
      href="#"
      type="button"
    >
      <svg
        class="delete-sign"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
        />
      </svg>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
